### PR TITLE
[#271] Getters for secure and insecure port in common base class.

### PR DESCRIPTION
**Pull request by sysexcontrol**
Eliminate code duplication from implementing the getters for each
specific vertx server (to only cover the ephemeral port determination).

Increase startup time for Hono Messaging in integration tests to 40 seconds.

Signed-off-by: Karsten Frank <Karsten.Frank@bosch-si.com>

### CODE REVIEW

File: tests/pom.xml [hunk 1]
```diff
@@ -252,7 +252,7 @@ Test cases are run against Docker images of Hono server + (Apache Qpid Dispatch
                     </log>
                     <wait>
                       <log>^.*(Started \w*Application in).*$</log>
-                      <time>30000</time>
+                      <time>40000</time>
```
Comment (sophokles73) [inline tests/pom.xml]: This has nothing to do with the other changes and should therefore go into a separate commit.

File: service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java [hunk 2]
```diff
@@ -182,6 +182,24 @@ public void registerLivenessChecks(final HealthCheckHandler handler) {
     public abstract int getInsecurePortDefaultValue();
 
     /**
+     * Gets the port number from a running secure server if it is listening on the corresponding socket already.
+     * <p>
+     * If no server is listening, {@link Constants#PORT_UNCONFIGURED} is returned.
+     *
+     * @return The port number.
+     */
+    protected abstract int getPortFromServer();
```
Comment (sophokles73) [inline service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java]: Can we rename this to `getActualPort` to match `NetServer.actualPort`?

File: service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java [hunk 3]
```diff
@@ -182,6 +182,24 @@ public void registerLivenessChecks(final HealthCheckHandler handler) {
...
+
+    /**
+     * Gets the port number from a running insecure server if it is listening on the corresponding socket already.
+     * <p>
+     * If no server is listening, {@link Constants#PORT_UNCONFIGURED} is returned.
+     *
+     * @return The port number.
+     */
+    protected abstract int getInsecurePortFromServer();
```
Comment (sophokles73) [inline service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java]: `getActualInsecurePort` or `getInsecureActualPort`?

File: tests/pom.xml (hunk 1 above)
Comment (sysexcontrol) [inline tests/pom.xml]: Oh yes, will move it out of this PR again (will need an increase though, often my environment needs about 31 seconds).

File: service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java (hunk 2 above)
Comment (sysexcontrol) [inline service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java]: +1

File: service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java (hunk 3 above)
Comment (sysexcontrol) [inline service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java]: getActualInsecurePort +1

File: tests/pom.xml (hunk 1 above)
Comment (sophokles73) [inline tests/pom.xml]: no need to remove it from the PR, just put it into a separate commit

**Comment by sysexcontrol** [comment]
Pushed timeout change as single commit on top (and won't squash it at after approval ;-)).

**Comment by sophokles73** [comment]
Is the second commit related to #271?

**Comment by sysexcontrol** [comment]
No, not directly. FMPOV it can be contained in another PR, or it remains in this PR but as a separate commit that remains visible in the master branch.
Bot are fine for me - WDYT?

**Comment by sophokles73** [comment]
No, I was just asking because in its commit message you are referring to #271 ...

**Comment by sysexcontrol** [comment]
Ok, I see - that is indeed misleading. Will change the commit message to not contain the PR number then.

**Comment by sophokles73** [comment]
Can you squash the first two commits?

**Comment by sysexcontrol** [comment]
Squashed first two commits, rebased branch.
